ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -874,8 +873,375 @@ ompi_report_comm_methods(int called_from_location) // 1 = from init, 2 = from fi
+    leader_comm->c_coll->coll_gather(
+        &len, 1, MPI_INT,
+        lens, 1, MPI_INT,
+        0, leader_comm, leader_comm->c_coll->coll_gather_module);
+    if (myleaderrank == 0) {
+        int tlen = 0;
+        char *p;
+        for (i=0; i<nleaderranks; ++i) {
+            disps[i] = tlen;
+            tlen += lens[i];
+        }
+        allhoststrings = malloc(nleaderranks * sizeof(char*)  +  tlen);
+        p = (char*) (allhoststrings + nleaderranks);
+        for (i=0; i<nleaderranks; ++i) {
+            allhoststrings[i] = p;
+            p += lens[i];
+        }
+        leader_comm->c_coll->coll_gatherv(
+            hoststring, strlen(hoststring) + 1, MPI_CHAR,
+            &allhoststrings[0][0], lens, disps, MPI_CHAR,
+            0, leader_comm, leader_comm->c_coll->coll_gatherv_module);
+    } else {
+        // matching above call from rank 0, just &allhoststrings[0][0]
+        // isn't legal here, and those args aren't used at non-root anyway
+        leader_comm->c_coll->coll_gatherv(
+            hoststring, strlen(hoststring) + 1, MPI_CHAR,
+            NULL, NULL, NULL, MPI_CHAR,
+            0, leader_comm, leader_comm->c_coll->coll_gatherv_module);
+    }
+    if (myleaderrank == 0) {
+        free(lens);
+        free(disps);
+    }
+
     if (myleaderrank == 0) {
+        for (i=0; i<nleaderranks; ++i) {
+            //printf("%s\n", allhoststrings[i]);
+            multiline_print(allhoststrings[i], cols, indentation);
+        }
         free(allhoststrings);
     }
-    free(method);
 }
+
+#include "opal/mca/hwloc/base/base.h"
+#define LENGTHEN(affstring, nextchar, affstring_cursz, amount) do { \
+    if ((int)(nextchar + amount + 16) >= (int)(*affstring_cursz)) { \
+        *affstring_cursz *= 1.25; \
+        *affstring_cursz += (amount + 16); \
+        *affstring = realloc(*affstring, *affstring_cursz); \
+    } \
+} while (0)
+
+/*
+ *  If the old --report-bindings output would have been this:
+ *            [BB/../../..][../../../..]
+ *            [../BB/../..][../../../..]
+ *            [../../../..][BB/../../..]
+ *            [../../../..][../BB/../..]
+ *  this function returns a string
+ *            [aa/bb/../..][cc/dd/../..]
+ *
+ *  This function decides how many ranks' bitmasks to process based on
+ *  1. avoiding overlap, and
+ *  2. only using 52 letters a-zA-Z
+ *  The *position argument should come in as 0 for the first call, and
+ *  it gets updated to point to the next host needing to be processed.
+ *
+ *  Each call to this function adds another line to an existing *affstring
+ *  (realloc()ing if needed).  The second-and-later lines get a carriage
+ *  return in front of them.  The last line does not end with a carriage
+ *  return.
+ */
+static void
+sprint_bitmaps(char **affstring, int *affstring_cursz,
+    int indentation,
+    hwloc_cpuset_t *cpus_forrank, int nranks,
+    int *position)
+{
+    int i, r;
+    int nchildren_done[32], depth, some_cores_printed_under_containing_obj;
```

Comment:
Probably should initialize `some_cores_printed_under_containing_obj` to 0.

Reply:
Agreed